From cd381974837d196271f4bbc1c607035f3a81f05c Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:33:08 -0400
Subject: [infiniband] qib: Fix port in pkey change event

Message-id: <bca9f4e3061c097d48d84619ea656bd5ebb4addb.1411579814.git.dledford@redhat.com>
Patchwork-id: 96689
O-Subject: [Patch RHEL7 15/19] IB/qib: Fix port in pkey change event
Bugzilla: 1085961
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Mike Marciniszyn <mike.marciniszyn@intel.com>

The code used a literal 1 in dispatching an IB_EVENT_PKEY_CHANGE.

As of the dual port qib QDR card, this is not necessarily correct.

Change to use the port as specified in the call.

Cc: <stable@vger.kernel.org>
Reported-by: Alex Estrin <alex.estrin@intel.com>
Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 911eccd284d13d78c92ec4f1f1092c03457d732a)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/qib/qib_mad.c b/drivers/infiniband/hw/qib/qib_mad.c
index e87a0f6..636be11 100644
--- a/drivers/infiniband/hw/qib/qib_mad.c
+++ b/drivers/infiniband/hw/qib/qib_mad.c
@@ -1028,7 +1028,7 @@ static int set_pkeys(struct qib_devdata *dd, u8 port, u16 *pkeys)
 
   event.event = IB_EVENT_PKEY_CHANGE;
   event.device = &dd->verbs_dev.ibdev;
-  event.element.port_num = 1;
+  event.element.port_num = port;
   ib_dispatch_event(&event);
  }
  return 0;
-- 
1.7.1